nistats/experimental_paradigm.py

```diff
@@ -46,21 +45,21 @@ def check_paradigm(paradigm):
         Per-event modulation, (in seconds)
         defaults to ones(n_events) when no duration is provided
     """
-    if 'onset' not in paradigm.keys():
-        raise ValueError('The provided paradigm has no onset key')
+    if 'onset' not in events.keys():
+        raise ValueError('The provided events have no onset key')
 
-    onset = np.array(paradigm['onset'])
+    onset = np.array(events['onset'])
     n_events = len(onset)
     trial_type = np.repeat('dummy', n_events)
     duration = np.zeros(n_events)
     modulation = np.ones(n_events)
-    if 'trial_type' in paradigm.keys():
-        warnings.warn("'trial_type' key not found in the given paradigm.")
-        trial_type = np.array(paradigm['trial_type'])
-    if 'duration' in paradigm.keys():
-        warnings.warn("'duration' key not found in the given paradigm.")
-        duration = np.array(paradigm['duration']).astype(np.float)
-    if 'modulation' in paradigm.keys():
-        warnings.warn("'modulation' key not found in the given paradigm.")
-        modulation = np.array(paradigm['modulation']).astype(np.float)
+    if 'trial_type' in events.keys():
```

Comment:
Oops, this is handled in PR #236 ...
Again, this one will have to go after #236 --which i am going to merge during the WE.